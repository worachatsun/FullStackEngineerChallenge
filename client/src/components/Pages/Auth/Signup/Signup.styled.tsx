import styled from "styled-components";

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
`;

export const Label = styled.div`
    margin-bottom: 20px;
    font-size: 1.5em;
`;
