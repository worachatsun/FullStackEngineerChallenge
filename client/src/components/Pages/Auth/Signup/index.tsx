import React, { FunctionComponent } from 'react';
import { Container, Label } from './Signup.styled';
import Input from '../../../commons/Input';
import Button from '../../../commons/Button';
import { useForm } from 'react-hook-form';
import { SignupSchema } from '../../../../schemas/signin';
import { SIGNUP_API, LIST_EMPLOYEES_API } from '../../../../constants/routes';
import { mutator, HttpMethod } from '../../../commons/utils/client';
import { useHistory } from 'react-router-dom';
import { mutate } from 'swr';

interface ISignup {
  username: string;
  password: string;
}

const Signup: FunctionComponent = () => {
  const { register, handleSubmit, errors } = useForm<ISignup>();
  let history = useHistory();

  const onSubmit = async (user: ISignup) => {
    try {
      const token = localStorage.getItem('token') || '';
      const { username, password } = user;
      const { response } = await mutator(
        SIGNUP_API,
        HttpMethod.POST,
        undefined,
        { username, password }
      );
      if (!response?.ok)
        throw new Error(`${response?.status}: ${response?.statusText}`);
      const data = await response?.json();
      mutate([LIST_EMPLOYEES_API, token]);
    } catch (error) {
      console.error(error);
    }
  };

  return (
    <Container>
      <form onSubmit={handleSubmit(onSubmit)}>
        <Label>Signup</Label>
        <Input
          title={'Username'}
          name='username'
          errors={errors.username}
          register={register}
        />
        <Input
          title={'Password'}
          name='password'
          errors={errors.password}
          register={register}
          type={'password'}
        />
        <Button>Signup</Button>
      </form>
    </Container>
  );
};

export default Signup;
