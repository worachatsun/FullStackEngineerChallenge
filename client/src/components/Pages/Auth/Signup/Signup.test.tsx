import React from "react";
import renderer from "react-test-renderer";
import Signup from "./index";

describe("<Signup />", () => {
    test("should render correctly", () => {
        const render = renderer.create(<Signup />).toJSON();
        expect(render).toMatchSnapshot();
    });
});
