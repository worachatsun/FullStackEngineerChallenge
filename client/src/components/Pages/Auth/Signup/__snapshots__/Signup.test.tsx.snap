// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> should render correctly 1`] = `
<div
  className="sc-AxiKw jAHgLp"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="sc-AxhCb bRcGML"
    >
      Signup
    </div>
    <div>
      <label
        htmlFor="Username"
      >
        Username
        :
        <input
          className="sc-AxirZ lmasuR"
          name="username"
          type="text"
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="Password"
      >
        Password
        :
        <input
          className="sc-AxirZ lmasuR"
          name="password"
          type="password"
        />
      </label>
    </div>
    <button
      className="sc-AxjAm fbqdBC"
    >
      Signup
    </button>
  </form>
</div>
`;
